src/main/java/io/aiven/kafka/connect/common/output/jsonwriter/JsonOutputWriter.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.jsonwriter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.output.OutputWriter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class JsonOutputWriter extends OutputWriter {
+
+    private final Map<String, OutputFieldBuilder> fieldBuilders;
+    private final ObjectMapper objectMapper;
+    private static final byte[] BATCH_START = "[\n".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] RECORD_SEPARATOR = ",\n".getBytes(StandardCharsets.UTF_8);
+    private static final byte[] BATCH_END = "\n]".getBytes(StandardCharsets.UTF_8);
+
+    JsonOutputWriter(final OutputStream outputStream,
+                     final Map<String, OutputFieldBuilder> fieldBuilders) {
+        super(outputStream);
+        this.fieldBuilders = fieldBuilders;
+        this.objectMapper = new ObjectMapper();
+        objectMapper.setNodeFactory(JsonNodeFactory.withExactBigDecimals(true));
+    }
+
+    @Override
+    protected void startWriting() throws IOException {
+        outputStream.write(BATCH_START);
+    }
+
+    protected void writeRecordsSeparator() throws IOException {
+        outputStream.write(RECORD_SEPARATOR);
+    }
+
+    protected void writeOneRecord(final SinkRecord record) throws IOException {
+        outputStream.write(objectMapper.writeValueAsBytes(getFields(record)));
+    }
+
+    @Override
+    public void close() throws IOException  {
+        outputStream.write(BATCH_END);
+    }
```

Comment:
A couple of notes about closing:
1. It's a good idea calling `outputStream.flush()` after the list write to ensure the data reaches the destination, because `close` (see next) is not always guaranteed to invoke `flush` automatically.
2. It's worth calling `outputStream.close()` here too. It's kind of a convention that the outer stream/writer closes its nested stream/writer on itself being closed.
3. It's strongly encouraged to make `close()` idempotent and not to perform any actions if it's already closed. Most of the implementations out there just use a private flag for this.

[`DeflaterOutputStream.close()`](https://github.com/JetBrains/jdk8u_jdk/blob/master/src/share/classes/java/util/zip/DeflaterOutputStream.java#L236) is a good example.

Reply:
Make sense

Reply:
- [ ] Implement AutoClosable